src/components/MoneyRequestAmountInput.tsx

```diff
@@ -154,25 +161,37 @@ function MoneyRequestAmountInput(
      * @param {String} newAmount - Changed amount from user input
      */
     const setNewAmount = useCallback(
-        (newAmount: string) => {
+        (newAmountInput: string) => {
+            const [newCurrencySymbol, newAmount = ''] = newAmountInput.split(/(\d.*)/, 2);
+            const newCurrency = CurrencyUtils.getCurrencyCode(newCurrencySymbol);
+
             // Remove spaces from the newAmount value because Safari on iOS adds spaces when pasting a copied value
             // More info: https://github.com/Expensify/App/issues/16974
             const newAmountWithoutSpaces = MoneyRequestUtils.stripSpacesFromAmount(newAmount);
+            const newDecimals = newCurrency ? CurrencyUtils.getCurrencyDecimals(newCurrency) : decimals;
             const finalAmount = newAmountWithoutSpaces.includes('.')
                 ? MoneyRequestUtils.stripCommaFromAmount(newAmountWithoutSpaces)
                 : MoneyRequestUtils.replaceCommasWithPeriod(newAmountWithoutSpaces);
+
+            // We should skip updating the amount if the currency symbol is invalid or we paste the amount with currency for only a part of amount
+            const shouldSkip = newCurrencySymbol && (!newCurrency || selection.end - selection.start < amount.length - 1);
+
             // Use a shallow copy of selection to trigger setSelection
             // More info: https://github.com/Expensify/App/issues/16385
-            if (!MoneyRequestUtils.validateAmount(finalAmount, decimals)) {
+            if (!MoneyRequestUtils.validateAmount(finalAmount, newDecimals) || shouldSkip) {
```

Comment:
Please rename the variable names as in https://github.com/Expensify/App/pull/47767#discussion_r1734950082

Reply:
I updated.